From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Robin Ebert <ebertrobin2002@gmail.com>
Date: Tue, 22 Feb 2022 18:12:22 +0100
Subject: Map keymap fd as read-only

Chromium tries to mmap the keymap file descriptor as read-write but it is only passed as read-only.

diff --git a/ui/ozone/platform/wayland/host/wayland_keyboard.cc b/ui/ozone/platform/wayland/host/wayland_keyboard.cc
index 272322210b59228e5faa606101957c48c6371249..84e30f9873c8bfdc9fb8486bcc955ca4a9fcad65 100644
--- a/ui/ozone/platform/wayland/host/wayland_keyboard.cc
+++ b/ui/ozone/platform/wayland/host/wayland_keyboard.cc
@@ -8,7 +8,7 @@
 
 #include "base/files/scoped_file.h"
 #include "base/logging.h"
-#include "base/memory/unsafe_shared_memory_region.h"
+#include "base/memory/read_only_shared_memory_region.h"
 #include "base/unguessable_token.h"
 #include "ui/base/buildflags.h"
 #include "ui/events/base_event_utils.h"
@@ -119,10 +119,10 @@ void WaylandKeyboard::Keymap(void* data,
   base::ScopedFD fd(keymap_fd);
   auto length = size - 1;
   auto shmen = base::subtle::PlatformSharedMemoryRegion::Take(
-      std::move(fd), base::subtle::PlatformSharedMemoryRegion::Mode::kUnsafe,
+      std::move(fd), base::subtle::PlatformSharedMemoryRegion::Mode::kReadOnly,
       length, base::UnguessableToken::Create());
   auto mapped_memory =
-      base::UnsafeSharedMemoryRegion::Deserialize(std::move(shmen)).Map();
+      base::ReadOnlySharedMemoryRegion::Deserialize(std::move(shmen)).Map();
   const char* keymap = mapped_memory.GetMemoryAs<char>();
 
   if (!keymap || format != WL_KEYBOARD_KEYMAP_FORMAT_XKB_V1)
